scheduler-server/scheduling-dsl-compiler/src/main.ts

```diff
@@ -1,22 +1,48 @@
 import fs from 'fs';
+import * as readline from 'readline';
 import ts from 'typescript';
 import { UserCodeRunner } from '@nasa-jpl/aerie-ts-user-code-runner';
-import type { Goal } from './libs/scheduler-edsl-fluent-api.js'
+import type { Goal } from './libs/scheduler-edsl-fluent-api.js';
 
 const codeRunner = new UserCodeRunner();
 const schedulerEDSL = fs.readFileSync(`${process.env.SCHEDULING_DSL_COMPILER_ROOT}/src/libs/scheduler-edsl-fluent-api.ts`, 'utf8');
 const schedulerAST = fs.readFileSync(`${process.env.SCHEDULING_DSL_COMPILER_ROOT}/src/libs/scheduler-ast.ts`, 'utf8');
 
+const input = readline.createInterface(process.stdin);
+
+function registerInputCallback(callback: (data: string) => void) {
+  input.once('line', callback);
```

Comment:
We might as well inject the handler here and have no parameters on this function